2996_reflection/reflection.md

```diff
@@ -3610,19 +3640,58 @@ Add another new paragraph defining _plainly constant-evaluated_ expressions:
     * during template argument deduction ([temp.deduct]{.sref}),
     * in a `$concept-id$` ([temp.names]{.sref}), or
     * in a `$requires-expression$` ([expr.prim.req]{.sref}), or
-  * [#.#.#]{.pnum} is a manifestly constant-evaluated initializer of a variable that is neither  `constexpr` ([dcl.constexpr]{.sref}) nor `constinit` ([dcl.constinit]{.sref}).
+  * [#.#.#]{.pnum} is, or is a subexpression of, an initializer for a variable that is neither  `constexpr` ([dcl.constexpr]{.sref}) nor `constinit` ([dcl.constinit]{.sref}).
+
+[As detailed below, the evaluation of a plainly constant-evaluated expression may produce new declarations. Their definition precludes expressions that may be evaluated repeatedly.]{.note12}
 
 :::
 :::
 
-Add new paragraphs defining _injected declarations_ and _injected points_:
+Modify the definition of _manifestly constant-evaluated_ to leverage that of _plainly constant-evaluated_:
```

Comment:
'm looking for clarity for like — what is the distinction between "plainly" and "manifestly" and when do either apply. I guess "plainly" is a strict subset of "manifestly", but what is the significance of "manifestly-but-not-plainly"?  

We currently use "manifestly" in a few places:

* How to take which branch in `if consteval` [here](https://eel.is/c++draft/stmt.if#4).
* How to define immediate function context [here](https://eel.is/c++draft/expr.const#16).
* As part of defining "needed for constant evaluation" [here](https://eel.is/c++draft/expr.const#21).

Should any of those be "plainly" instead?

Reply:
"manifestly-but-not-plainly" implies: This expression will (with caveats) be evaluated during translation, but may be evaluated multiple times (and we make no guarantees around the order of evaluation during translation).

Reply:
This is great. Can you add notes to the end of plainly and manifestly that explain in quick layman's terms what they mean? What you said above for manifestly-but-not-plainly is great.

Reply:
Pretty sure we don't want to change any of the three links cited.

Reply:
> This is great. Can you add notes to the end of plainly and manifestly that explain in quick layman's terms what they mean? What you said above for manifestly-but-not-plainly is great.

Done.